RFCs/0002-pluggable-discovery.md

```diff
@@ -0,0 +1,568 @@
+# Pluggable Discovery protocol (Version 1.0.0)
+
+## Overview
+
+This document describes how the Pluggable Discovery works and how it should integrate with monitors and uploaders.
+
+## Problem
+
+When a sketch is uploaded to an Arduino board the only way for transferring the binary executable to the microcontroller is through the serial port. Currently:
```

Comment:
This RFC started as documentation of the current pluggable discovery support in the Java IDE: ideally, once done, this document will look like it was done before the implementation in the Java IDE.

BTW I quickly realized that this specification needs to cover a lot more cases that are currently not handled in the Java IDE so, I decided to not document the current status of pluggable discovery in the Java IDE because it will complicate the specification. The only "backward compatibility" I want to support is related to the classic non-pluggable upload, which covers 99% of the current uploads implemented in the wild.

This will require more work from Paul, as an early adopter, to provide some backward compatibility (and I'm glad to help on this side) but will help produce cleaner documentation that pays in the long term IMHO.

Reply:
> This will require more work from Paul, as an early adopter, to provide some backward compatibility

The HELLO message is all I need.  :)